specutils/manipulation/extract_spectral_region.py

```diff
@@ -75,7 +75,19 @@ def _subregion_to_edge_pixels(subregion, spectrum):
         if not spectral_axis.unit.is_equivalent(u.pix):
             left_index = floor(subregion[0].value)
         else:
-            left_index = int(np.ceil(left_func(subregion).value))
+            if order != "ascending":
+                raise ValueError("u.pix handling should always be ascending")
+            # If the lower bound is larger than the largest value, immediately return nothing
+            # Assuming ascending, both bounds are "out of bounds"
+            if subregion[0] > spectral_axis[-1]:
+                return None, None
+            else:
+                # Get index of closest value
+                # See https://stackoverflow.com/a/26026189 if performance becomes an issue
+                left_index = np.nanargmin((np.abs(spectral_axis - subregion[0])))
```

Comment:
Are there any cases where the spectral axis values would be NaN? That seems like something we should disallow in general.

Reply:
I dont' think there would be any "intended"/"expected" situations where there would be a NaN in the spectral axis, but unless there are preexisting checks that explicitly forbid this from happening, the default behavior of `np.argmin` to always select the `nan` is quite concerning.

Also, if we can futureproof against that case, then I say why not? `np.nanargmin` vs `np.argmin` is just one small substitution